api/v1beta1/awsmachine_types.go

```diff
@@ -179,6 +186,15 @@ type CloudInit struct {
 	SecureSecretsBackend SecretBackend `json:"secureSecretsBackend,omitempty"`
 }
 
+// Ignition defines options related to the bootstrapping systems where Ignition is used.
+type Ignition struct {
+	// Version defines which version of Ignition will be used to generate bootstrap data.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum="2.3"
+	Version string `json:"version,omitempty"`
```

Comment:
>How about adding a default value here and making this field required, instead of setting the default in the defaulting webhook?
IMO, it'd make changing the default and adding a new enum all at one place.

Sure, make sense. Though I see:
```
  // Scheme sets the scheme of the load balancer (defaults to internet-facing)
  // +kubebuilder:default=internet-facing
  // +kubebuilder:validation:Enum=internet-facing;internal
  // +optional
  Scheme *ClassicELBScheme `json:"scheme,omitempty"`
```

Which is similar to this case. Do we want optional + default or is the existing type annotation wrong?

Reply:
It is still optional from the user perspective, user does not need to provide it. But defaulted to this value by the webhook. So this looks okay.

Reply:
So `+kubebuilder:default="2.3"` is now added and optional tag remains. As far as I understand, this is addressed, so I'll mark it as resolved. Thanks for clarification.